api/src/constants.ts

```diff
@@ -104,3 +104,5 @@ export const RESUMABLE_UPLOADS = {
 	EXPIRATION_TIME: getMilliseconds(env['TUS_UPLOAD_EXPIRATION'], 600_000 /* 10min */),
 	SCHEDULE: String(env['TUS_CLEANUP_SCHEDULE'] as string),
 };
+
+export const ALLOWED_DB_DEFAULT_FUNCTIONS = ['gen_random_uuid()'];
```

Comment:
Any reason why we don't just expose a new configuration variable like `DB_ALLOWED_DEFAULT_FUNCTIONS`? Would add more flexibility for the user, we don't have to maintain a list of common functions and can even leave it empty by default

Reply:
mainly because it would take more time to implement and I don't think this is a requested feature. it would also mean that the user potentially runs into an error for the first schema migration (or field creation), so this seems more inconvenient for the user. maybe for directus next? otherwise feel free to implement it, but we only have next week to fix some other important server-side issues

Reply:
it may even introduce a vulnerability when an attacker overwrites the config 🤔

Reply:
I agree that this is sufficient for now 👍 If there is demand for it, we can consider making it configurable later on or look into https://github.com/directus/directus/pull/23359#discussion_r1718683753 again.

Reply:
> it may even introduce a vulnerability when an attacker overwrites the config 🤔

If an attacker is able to overwrite a config, I thing security is a thing of the past anyways 😃